### PR TITLE
FE-1367: upload validation errors and warnings to validate job artifacts

### PR DESCRIPTION
**Pull request by tremarkley**
With this change any warnings or errors that occur during validation will be uploaded to the artifacts of the `validate` job

### CODE REVIEW

**Comment by tremarkley** [comment]
Current dependencies on/for this PR:
* master
  * **PR #315** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ethereum-optimism.github.io/315" target="_blank">Graphite</a> 
    * **PR #317** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ethereum-optimism.github.io/317" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/ethereum-optimism.github.io/317?utm_source=stack-comment).

**Comment by tremarkley** [comment]
Do not merge this pull request until a security review has been conducted on it

File: .github/workflows/validate.yml
Comment (roninjin10) [inline .github/workflows/validate.yml:80]:
```suggestion
        if: failure()
```
It can be only on failures right?

**Review by roninjin10 — APPROVED** [review]

File: .github/workflows/validate.yml
Comment (tremarkley) [reply .github/workflows/validate.yml:80]: now that I combined this to one file this will have to be set to always because warnings will not trigger a failure

**Comment by tremarkley** [comment]
waiting to merge this until @zchn reviews

File: bin/cli.ts
Comment (roninjin10) [inline bin/cli.ts:72]: nit unrelated to pr:

Would prefer if we just throw errors and then on the most outside external part of the code have some logic like

runValidaiton().catch(e => console.error('there was a problem', e) exit(1))

**Review by roninjin10 — APPROVED** [review]

**Pull request merged**
